flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBIncrementalCheckpointUtils.java

```diff
@@ -154,6 +162,54 @@ private static void deleteRange(
         }
     }
 
+    /**
+     * Clip the entries in the CF according to the range [begin_key, end_key). Any entries outside
+     * this range will be completely deleted (including tombstones).
+     *
+     * @param db the target need to be clipped.
+     * @param columnFamilyHandles the column family need to be clipped.
+     * @param beginKeyBytes the begin key bytes
+     * @param endKeyBytes the end key bytes
+     */
+    public static void clipColumnFamilies(
+            RocksDB db,
+            List<ColumnFamilyHandle> columnFamilyHandles,
+            byte[] beginKeyBytes,
+            byte[] endKeyBytes)
+            throws RocksDBException {
+
+        for (ColumnFamilyHandle columnFamilyHandle : columnFamilyHandles) {
+            db.clipColumnFamily(columnFamilyHandle, beginKeyBytes, endKeyBytes);
+        }
+    }
+
+    public static List<Pair<RegisteredStateMetaInfoBase, ExportImportFilesMetaData>>
+            exportColumnFamilies(
+                    RocksDB db,
+                    List<ColumnFamilyHandle> columnFamilyHandles,
+                    List<StateMetaInfoSnapshot> stateMetaInfoSnapshots,
```

Comment:
This to lists should match each other, right?
Should we check with `Preconditions.checkState(columnFamilyHandles.size() == stateMetaInfoSnapshots.size());`,
or better pass RocksDB instance here?